pyvrp/cpp/ProblemData.cpp

```diff
@@ -163,6 +163,16 @@ ProblemData::ProblemData(std::vector<Client> const &clients,
             throw std::invalid_argument("Depot release time must be 0.");
     }
 
+    // Check that diagonal of distance and duration matrix is equal to 0.
+    for (size_t i = 0; i < numLocations(); ++i)
+    {
+        if (dist_(i, i) != 0)
+            throw std::invalid_argument("Distance matrix diagonal must be 0.");
+
+        if (dur_(i, i) != 0)
+            throw std::invalid_argument("Duration matrix diagonal must be 0.");
+    }
```

Comment:
```suggestion
    for (size_t idx = 0; i != numLocations(); ++idx)
    {
        if (dist_(idx, idx) != 0)
            throw std::invalid_argument("Distance matrix diagonal must be 0.");

        if (dur_(idx, idx) != 0)
            throw std::invalid_argument("Duration matrix diagonal must be 0.");
    }
```

Reply:
I'd move this up a little bit, to just below the matrix dimension checks. Then all "matrix stuff" is nicely grouped together.